### PR TITLE
[github-actions] build docker image for reference device

### PR DESCRIPTION
**Pull request by simonlingoogle**
This PR enhance the docker build test to use more options. 

**This PR also builds and pushes openthread/otbr:reference-device** for certification purpose.

### CODE REVIEW

**Comment by simonlingoogle** [comment]
@librasungirl This is a tiny PR that adds more build flags to docker buldx job for testing purpose. 
Please take a look when you have time.

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -46,8 +46,11 @@ jobs:
       matrix:
         include:
           - build_args: ""
+            image_tag: "latest"
+          - build_args: "--build-arg REFERENCE_DEVICE=1 --build-arg BACKBONE_ROUTER=1 --build-arg NAT64=0 --build-arg OTBR_OPTIONS='-DOT_DUA=ON -DOT_MLR=ON'"
```
Comment (librasungirl) [inline .github/workflows/docker.yml:50]: Build `latest` with 1.2 relative feature flags by default now?

Comment (simonlingoogle) [reply .github/workflows/docker.yml:50]: Note That the latest tag does not use any 1.2 feature. 
Btw, I changed the purpose of this PR. Labeled as wip.

**Comment by simonlingoogle** [comment]
The docker build checks are passed now. @librasungirl @jwhui

**Review by librasungirl — APPROVED** [review]

**Comment by simonlingoogle** [comment]
I think we can't pass all docker buildx checks because some checks are adjusted. @jwhui

File: .github/workflows/docker.yml [hunk 2]
```diff
@@ -64,18 +70,18 @@ jobs:
...
         DOCKER_PLATFORMS=linux/amd64,linux/arm/v7,linux/arm64
-        VERSION=latest
+        VERSION=${{ matrix.image_tag }}
 
         TAGS="--tag ${DOCKER_IMAGE}:${VERSION}"
 
         echo ::set-output name=docker_image::${DOCKER_IMAGE}
         echo ::set-output name=version::${VERSION}
-        echo ::set-output name=buildx_args::--platform ${DOCKER_PLATFORMS} \
+        echo ::set-output name=buildx_args::"--platform ${DOCKER_PLATFORMS} \
```
Comment (jwhui) [inline .github/workflows/docker.yml:79]: Are the quotes necessary here?

Comment (simonlingoogle) [reply .github/workflows/docker.yml:79]: Yes. It makes sure `--build-arg OTBR_OPTIONS='-DOT_DUA=ON -DOT_MLR=ON'` is correctly set to the `buildx_args` output.

**Review by jwhui — APPROVED** [review]

**Pull request merged**